### PR TITLE
[SPARK-25031][SQL] Fix MapType schema print

### PR DESCRIPTION
**Pull request by invkrh**
## What changes were proposed in this pull request?

The PR fix the bug in `buildFormattedString` function in `MapType`, which makes the printed schema misleading.

## How was this patch tested?

Added UT

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/types/MapTypeSuite.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/*
```
Comment (mgaido91) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/types/MapTypeSuite.scala]: I think we can put this in `DataTypeSuite`

**Review by mgaido91 — COMMENTED** [review]
the change itself LGTM, just left one comment.

cc @gatorsmile @HyukjinKwon for triggering the build

**Comment by HyukjinKwon** [comment]
ok to test

File: sql/catalyst/src/test/scala/org/apache/spark/sql/types/MapTypeSuite.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+package org.apache.spark.sql.types
+
+import org.apache.spark.SparkFunSuite
+
+class MapTypeSuite extends SparkFunSuite {
+  test("SPARK-25031") {
+
+    val keyType: DataType = StructType(
+      Seq(StructField("a", DataTypes.IntegerType),
+          StructField("b", DataTypes.IntegerType)))
```
Comment (HyukjinKwon) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/types/MapTypeSuite.scala]: nit: 2 space indentation

File: sql/catalyst/src/test/scala/org/apache/spark/sql/types/MapTypeSuite.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.types
+
+import org.apache.spark.SparkFunSuite
+
+class MapTypeSuite extends SparkFunSuite {
+  test("SPARK-25031") {
```
Comment (HyukjinKwon) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/types/MapTypeSuite.scala]: nit: `SPARK-25031: MapType should produce current formatted string for complex types`

**Comment by HyukjinKwon** [comment]
LGTM too otherwise

File: sql/catalyst/src/test/scala/org/apache/spark/sql/types/MapTypeSuite.scala (hunk 2 above)
Comment (invkrh) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/types/MapTypeSuite.scala]: Done

File: sql/catalyst/src/test/scala/org/apache/spark/sql/types/MapTypeSuite.scala (hunk 3 above)
Comment (invkrh) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/types/MapTypeSuite.scala]: Done

**Comment by mgaido91** [comment]
the test failure is flaky. Just created a PR for fixing it: https://github.com/apache/spark/pull/22023. I'd still prefer to merge the added UT to `DataTypeSuite` as all UTs regarding the types are there as of now. But if you @HyukjinKwon  think we can use the new test suite, then we can retrigger the tests. Thanks.

**Comment by HyukjinKwon** [comment]
Let's just put this in `DataTypeSuite` since we need another test that should be ran anyway.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/types/DataTypeSuite.scala [hunk 4]
```diff
@@ -452,4 +452,31 @@ class DataTypeSuite extends SparkFunSuite {
     new StructType().add("f1", IntegerType).add("f", new StructType().add("f2", StringType, false)),
     new StructType().add("f2", IntegerType).add("g", new StructType().add("f1", StringType)),
     false)
+
+  test("SPARK-25031: MapType should produce current formatted string for complex types") {
+
```
Comment (mgaido91) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/types/DataTypeSuite.scala]: nit: unneeded blank line

**Review by mgaido91 — APPROVED** [review]
LGTM

File: sql/catalyst/src/test/scala/org/apache/spark/sql/types/DataTypeSuite.scala (hunk 4 above)
Comment (invkrh) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/types/DataTypeSuite.scala]: Done

**Comment by mgaido91** [comment]
retest this please

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Pull request closed without merging**